homeassistant/components/konnected.py

```diff
@@ -0,0 +1,298 @@
+        configured_devices = self.user_config[CONF_DEVICES]
+        return next((device for device in
+                     configured_devices if device[CONF_ID] in valid_keys),
+                    None)
+
+    def save_data(self):
+        """Save the device configuration to `hass.data`.
+
+        TODO: This can probably be refactored and tidied up.
+        """
+        sensors = {}
+        for entity in self.config().get(CONF_SENSORS) or []:
+            if CONF_ZONE in entity:
+                pin = ZONE_TO_PIN[entity[CONF_ZONE]]
+            else:
+                pin = entity[CONF_PIN]
+
+            sensor_status = next((sensor for sensor in
+                                  self.status.get('sensors') if
+                                  sensor.get(CONF_PIN) == pin), {})
+            if sensor_status.get(ATTR_STATE):
+                initial_state = bool(int(sensor_status.get(ATTR_STATE)))
+            else:
+                initial_state = None
+
+            sensors[pin] = {
+                CONF_TYPE: entity[CONF_TYPE],
+                CONF_NAME: entity.get(CONF_NAME, 'Konnected {} Zone {}'.format(
+                    self.device_id[6:], PIN_TO_ZONE[pin])),
+                ATTR_STATE: initial_state
+            }
+            _LOGGER.info('Set up sensor %s (initial state: %s)',
+                         sensors[pin].get('name'),
+                         sensors[pin].get(ATTR_STATE))
+
+        actuators = {}
+        for entity in self.config().get(CONF_SWITCHES) or []:
+            if 'zone' in entity:
+                pin = ZONE_TO_PIN[entity['zone']]
+            else:
+                pin = entity['pin']
+
+            actuator_status = next((actuator for actuator in
+                                    self.status.get('actuators') if
+                                    actuator.get('pin') == pin), {})
+            if actuator_status.get(ATTR_STATE):
+                initial_state = bool(int(actuator_status.get(ATTR_STATE)))
+            else:
+                initial_state = None
+
+            actuators[pin] = {
+                CONF_NAME: entity.get(
+                    CONF_NAME, 'Konnected {} Actuator {}'.format(
+                        self.device_id[6:], PIN_TO_ZONE[pin])),
+                ATTR_STATE: initial_state,
+                'activation': entity['activation'],
+            }
+            _LOGGER.info('Set up actuator %s (initial state: %s)',
+                         actuators[pin].get(CONF_NAME),
+                         actuators[pin].get(ATTR_STATE))
+
+        device_data = {
+            'client': self.client,
+            CONF_SENSORS: sensors,
+            CONF_SWITCHES: actuators,
+            CONF_HOST: self.host,
+            CONF_PORT: self.port,
+        }
+
+        if 'devices' not in self.hass.data[DOMAIN]:
+            self.hass.data[DOMAIN][CONF_DEVICES] = {}
+
+        _LOGGER.info('Storing data in hass.data[konnected]: %s', device_data)
```

Comment:
This also seems like a non-info log message. Especially with `info` messages, I don't expect to see raw python dictionaries in the logs.